src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2750,188 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
+        # HEADER
+        # key = value unit
+
+        # Example:
+        # AP_STAGE_AT_X
+        # Stage at X = 48.5090 mm
+
+        try:
+            
+            # skip all caps lines as they are headers
+            if line.isupper():
+                continue
+
+            # NOTE: 
+            # Date, Time are separated by : 
+            # everything else is separated by =
+            try:
+                k, v = line.split("=")
+            except ValueError:             
+                k, v = line.split(":", maxsplit=1) 
+            k, v = k.strip(), v.strip()
+            zmd[k] = v
+        except ValueError:
+            continue  # skip lines that don't contain key value pairs
+        except Exception as e:
+            # any other error -> something has gone wrong
+            logging.warning(f"Failed to parse ZEISS metadata: {line} - {e}", exc_info=True)
+
+    return zmd
+
+
+def _convert_zeiss_to_odemis_metadata(metadata: str) -> dict:
+    """Converts ZEISS metadata to odemis metadata format
+    :param metadata: metadata string
+    :return: (dict) metadata as dictionary
+    """
+
+    # parse as dict
+    # from pprint import pprint
+    zmd = _parse_zeiss_metadata(metadata)
+
+    md = {}
+
+    # general
+    try:
+        datetime_str = zmd["Date"] + " " + zmd["Time"]
+        date = datetime.strptime(datetime_str, "%d %b %Y %H:%M:%S")
+        md[model.MD_ACQ_DATE] = date.timestamp()
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_ACQ_DATE} - {e}")
+    try:
+        md[model.MD_HW_NAME] = zmd["Serial No."]
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_HW_NAME} -  {e}")
+    try:
+        md[model.MD_SW_VERSION] = zmd["Version"]
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_SW_VERSION} - {e}")
+
+    try:
+        # NOTE: 
+        # There are three values in the ZEISS metadata for pixel size
+        # Pixel Size, Image Pixel Size, and FIB Pixel Size
+        # Pixel Size is the one we want as it changes based on the beam type (FIB Imaging field)
+
+        str_pixel_size = zmd["Pixel Size"]
+        str_val, si_prefix, unit = units.decompose_si_prefix(str_pixel_size, "m")
+        pixel_size = units.si_scale_val(float(str_val), si_prefix)
+
+        md[model.MD_PIXEL_SIZE] = (pixel_size, pixel_size)
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_PIXEL_SIZE} - {e}")
+
+    try:
+        str_dwell_time = zmd["Dwell Time"]
+        str_val, si_prefix, unit = units.decompose_si_prefix(str_dwell_time, "s")
+        dwell_time = units.si_scale_val(float(str_val), si_prefix)
+
+        md[model.MD_DWELL_TIME] = dwell_time
+    except Exception as e:
+        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_DWELL_TIME} - {e}")
+    try:
+
+        str_x = zmd["Stage at X"]
```

Comment:
can you move the empty line to above the `try:`?
```suggestion
        logging.warning(f"Failed to parse ZEISS metadata: {model.MD_DWELL_TIME} - {e}")

    try:
        str_x = zmd["Stage at X"]
```